core/src/com/po/fuck/model/weapons/Gun.java

```diff
@@ -34,19 +30,20 @@ public boolean attack() {
             return false;
 
         Vector2 direction = getDirection(), gunPosition = getPosition();
-
-//<<<<<<< HEAD:core/src/com/po/fuck/weapons/Gun.java
         gunPosition.add(new Vector2(geometryData.getWidth() / 2, 0).rotateRad(direction.angleRad()));
-        Vector2 realMuzzlePosition = new Vector2(muzzlePosition);
-        realMuzzlePosition.x -= geometryData.getWidth();
-        realMuzzlePosition.y -= geometryData.getHeight()/2;
-        Vector2 finalPosition = GeometryMisc.getPointPositionOnFlippedSprite(gunPosition.cpy(), direction,
-                realMuzzlePosition, direction.angleDeg() >= 90 && direction.angleDeg() <= 270);
+
+        JsonValue info = Assets.getInfo(name);
+        Vector2 realMuzzlePosition = new Vector2((float) info.getDouble("muzzlePositionX"),
```

Comment:
Why are you calculating this, each `attack` ?

Reply:
Why are you calculating `muzzlePosition` each attack and not in the class attributes ?

Reply:
@RyhorKoshman why are you ignoring this ?